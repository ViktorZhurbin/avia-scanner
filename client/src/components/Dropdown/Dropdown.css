.container {
    display: flex;
    flex-direction: column;
    position: relative;
}

.trigger {
    outline: 0;
}

.formSelect {
    flex-grow: 1;
    flex-basis: 33%;
}
