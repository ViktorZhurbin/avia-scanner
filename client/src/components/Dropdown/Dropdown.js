import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import OutsideClickHandler from 'react-outside-click-handler';
import cl from 'classnames/bind';

import { classNamesPropType } from '../../entities/propTypes';

import styles from './Dropdown.css';

const cx = cl.bind(styles);

const Dropdown = ({
    trigger,
    children,
    classNames,
}) => {
    const [isOpen, setDropdownState] = useState(false);

    const toggleDropdown = () => setDropdownState(!isOpen);

    const onOutsideClick = () => setDropdownState(false);

    const handleKeyPress = (event) => {
        if (event.key === 'Enter') {
            toggleDropdown();
        }
    };

    return (
        <div
            className={cx({
                container: true,
                ...classNames,
            })}
            role="button"
            tabIndex="0"
            onClick={toggleDropdown}
            onKeyPress={handleKeyPress}
        >
            <OutsideClickHandler
                onOutsideClick={onOutsideClick}
            >
                <div
                    className={cx('trigger')}
                    role="button"
                    tabIndex="0"
                    onClick={toggleDropdown}
                    onKeyPress={handleKeyPress}
                >
                    {trigger}
                </div>
                <CSSTransition
                    in={isOpen}
                    timeout={10}
                    classNames="message"
                    unmountOnExit
                >
                    <div className={cx('dropdown')}>
                        {children}
                    </div>
                </CSSTransition>
            </OutsideClickHandler>
        </div>
    );
};

Dropdown.propTypes = {
    trigger: PropTypes.node.isRequired,
    children: PropTypes.node.isRequired,
    classNames: classNamesPropType,
};

Dropdown.defaultProps = {
    classNames: null,
};

export default Dropdown;
