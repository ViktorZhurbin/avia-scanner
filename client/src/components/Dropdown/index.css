.container {
    display: flex;
    flex-direction: column;
    position: relative;
}

.trigger {
    outline: 0;
}

.dropdown {
    z-index: 11;
}

.placeSelect {
    flex-grow: 1;
    flex-basis: 33%;
}
