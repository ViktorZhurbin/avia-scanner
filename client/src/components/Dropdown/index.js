import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import ReactOutsideEvent from 'react-outside-event';
import cl from 'classnames/bind';

import { classNamesPropType } from '../../entities/propTypes';

import styles from './index.css';

const cx = cl.bind(styles);

class Dropdown extends React.PureComponent {
    static propTypes = {
        trigger: PropTypes.node.isRequired,
        children: PropTypes.node.isRequired,
        classNames: classNamesPropType,
    }

    static defaultProps = {
        classNames: null,
    }

    state = {
        isOpen: false,
    }

    onToggle = () => {
        const { isOpen } = this.state;

        this.setState(() => ({
            isOpen: !isOpen,
        }));
    }

    onOutsideEvent = () => {
        this.setState(() => ({
            isOpen: false,
        }));
    }

    renderDropdown = () => {
        const { isOpen } = this.state;
        const { children } = this.props;

        return (
            <CSSTransition
                in={isOpen}
                timeout={10}
                classNames="message"
                unmountOnExit
            >
                <div className={cx('dropdown')}>
                    {children}
                </div>
            </CSSTransition>
        );
    }

    handleKeyPress = (event) => {
        if (event.key === 'Enter') {
            this.onToggle();
        }
    }

    render() {
        const { trigger, classNames } = this.props;

        return (
            <div
                className={cx({
                    container: true,
                    ...classNames,
                })}
                role="button"
                data-toggle="dropdown"
                tabIndex="0"
                onClick={this.onToggle}
                onKeyPress={this.handleKeyPress}
            >
                <div
                    className={cx('trigger')}
                    role="button"
                    data-toggle="dropdown"
                    tabIndex="0"
                    onClick={this.onToggle}
                    onKeyPress={this.handleKeyPress}
                >
                    {trigger}
                </div>
                {this.renderDropdown()}
            </div>
        );
    }
}

export default ReactOutsideEvent(Dropdown, ['click']);
