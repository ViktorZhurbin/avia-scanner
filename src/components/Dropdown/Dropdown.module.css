.container {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    position: relative;
}

.dropdown {
    position: absolute;
    top: 42px;
    right: 0;
    width: max-content;
    box-shadow: 0 0 15px rgba(0,0,0,.25);
    margin-top: 10px;
    padding: 15px 0;
    background: white;
    border-radius: 5px;
}
