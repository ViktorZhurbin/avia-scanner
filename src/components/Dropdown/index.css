.container {
    display: flex;
    flex-direction: column;
    position: relative;
}

.trigger {
    outline: 0;
}

.dropdown {
    z-index: 11;
}
