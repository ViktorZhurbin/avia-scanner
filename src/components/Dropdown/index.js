import React from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import cl from 'classnames/bind';

import styles from './index.css';

const cx = cl.bind(styles);

class Dropdown extends React.Component {
    static propTypes = {
        trigger: PropTypes.node.isRequired,
        children: PropTypes.node.isRequired,
    }

    state = {
        isOpen: false,
    }

    onToggle = () => {
        const { isOpen } = this.state;

        this.setState(() => ({
            isOpen: !isOpen,
        }));
    }

    renderDropdown = () => {
        const { isOpen } = this.state;
        const { children } = this.props;

        return (
            <CSSTransition
                in={isOpen}
                timeout={100}
                classNames="message"
                unmountOnExit
            >
                <div className={cx('dropdown')}>
                    {children}
                </div>
            </CSSTransition>
        );
    }

    handleKeyPress = (event) => {
        if (event.key === 'Enter') {
            this.onToggle();
        }
    }

    render() {
        const { trigger } = this.props;

        return (
            <div
                className={cx('container')}
                role="button"
                data-toggle="dropdown"
                tabIndex="0"
                onClick={this.onToggle}
                onKeyPress={this.handleKeyPress}
            >
                <div
                    className={cx('trigger')}
                    role="button"
                    data-toggle="dropdown"
                    tabIndex="0"
                    onClick={this.onToggle}
                    onKeyPress={this.handleKeyPress}
                >
                    {trigger}
                </div>
                {this.renderDropdown()}
            </div>
        );
    }
}

export default Dropdown;
