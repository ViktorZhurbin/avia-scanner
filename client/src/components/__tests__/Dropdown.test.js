import React from 'react';
import { shallow } from 'enzyme';

import Dropdown from '../Dropdown';

const props = {
    onSelect: () => null,
};

describe('Render Dropdown', () => {
    it('render correctly dropdown component', () => {
        const component = shallow(
            <Dropdown
                {...props}
                trigger={<div />}
            >
                <ul>
                    <li>One</li>
                    <li>Two</li>
                </ul>
            </Dropdown>,
        );
        expect(component).toMatchSnapshot();
    });
});
