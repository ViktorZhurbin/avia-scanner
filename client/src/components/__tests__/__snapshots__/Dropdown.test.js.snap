// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Dropdown render correctly dropdown component 1`] = `ShallowWrapper {}`;
